historyarchive/archive_pool.go

```diff
@@ -51,8 +51,18 @@ func NewArchivePool(archiveURLs []string, opts ArchiveOptions) (ArchivePool, err
 	return validArchives, nil
 }
 
+func (pa ArchivePool) GetStats() []ArchiveStats {
+	stats := []ArchiveStats{}
```

Comment:
we can pre-allocate the known capacity here, yeah?
```suggestion
	stats := make([]ArchiveStats, 0, len(pa))
```

Reply:
I think potential capacity on stats isn't [one-to-one irt archive](https://github.com/stellar/go/pull/5183/commits/98d54d1623e2bd3f1370061b40fb04aa53dcc03c), there could be more than one stats instance, not at present, it doesn't seem like big gain vs. code either way?